File: internal/buildengine/languageplugin/testdata/java/plugintest/src/main/java/xyz/block/ftl/java/test/plugintest/PluginTest.java
package xyz.block.ftl.java.test.plugintest;

import xyz.block.ftl.Verb;

// uncommentForDependency:import ftl.dependable.Data;

public class PluginTest {
    // uncommentForDependency: Data data;
    @Verb
    public EchoResponse verbaabbcc(EchoRequest req) {

        EchoResponse rr = new EchoResponse();
        rr.message = String.format("Hello, %s!", req.name.orElse(("anonymous")));
        return rr;
    }

}

File: internal/buildengine/languageplugin/testdata/go/plugintest/plugintest.go
package plugintest

import (
	"context"
	"fmt"

	"github.com/block/ftl/go-runtime/ftl" // Import the FTL SDK.
	// uncommentForDependency: "ftl/dependable"
)

type EchoRequest struct {
	Name ftl.Option[string] `json:"name"`
}

type EchoResponse struct {
	Message string `json:"message"`
}

//ftl:verb
func Verbaabbcc(ctx context.Context, req EchoRequest) (EchoResponse, error) {
	// uncommentForDependency: fmt.Printf("printing %v", dependable.Data{})
	return EchoResponse{Message: fmt.Sprintf("Hello, %s!", req.Name.Default("anonymous"))}, nil
}
